kube-derive/src/custom_resource.rs

```diff
@@ -512,6 +518,18 @@ pub(crate) fn derive(input: proc_macro2::TokenStream) -> proc_macro2::TokenStrea
         quote! {}
     };
 
+    let deprecation = if let Some(deprecation) = deprecated {
+        match deprecation {
+            Override::Inherit => quote! { "deprecated": true, },
+            Override::Explicit(warning) => quote! {
+                "deprecated": true,
+                "deprecationWarning": #warning,
+            },
```

Comment:
i like this. it mirrors the behaviour of [rust's deprecated macro](https://doc.rust-lang.org/reference/attributes/diagnostics.html?highlight=depreca#the-deprecated-attribute) so the magic here is understandable.

Reply:
Yes, Rust's attribute served as the inspiration.